core-kotlin-modules/core-kotlin-collections-6/src/test/kotlin/com/baeldung/parallelOperationsCollections/ParallelOperationCollectionsUnitTest.kt

```diff
@@ -159,28 +160,5 @@ class ParallelOperationCollectionsUnitTest {
 
         filteredPeople.assertOver15AndSortedByAge()
     }
-
-    @Test
-    fun `using ExecutorService for parallel operations`() {
-        logger.info("Using ExecutorService")
-        val startTime = Instant.now()
-
-        val executor = Executors.newFixedThreadPool(people.size)
-        val futures = people
-            .map { person ->
-                executor.submit(Callable {
-                    person.setAdult()
-                    person
-                }).get()
-            }
-            .filter { it.age > 15 }
-            .sortedBy { it.age }
-
-        executor.shutdown()
-
-        startTime.printTotalTime()
-
-        futures.assertOver15AndSortedByAge()
-    }
```

Comment:
We could potentially show a solution where we collect a `List<Future<T>>` and transform that to a `List<T>`

Any reason for not showing this kind of solution?

Reply:
What does it mean to be like this?

        val executor = Executors.newFixedThreadPool(people.size)
        val futures: List<Future<Person>> = people
            .map { person ->
                executor.submit(Callable {
                    person.setAdult()
                    person
                })
            }

        // List<Future<Person>> to List<Person>
        val results = futures
            .map { it.get() }
            .filter { it.age > 15 }
            .sortedBy { it.age }

        executor.shutdown()

        results.assertOver15AndSortedByAge()

Reply:
@hangga yes that's what I had in mind. I've used an approach similar to this in the past in plain Java.

Reply:
After you agreed, I am more confident now. Because before, I was not sure about what I was doing.